### PR TITLE
Adds INDRA Reactor Alarm

### PR DESCRIPTION
**Pull request by CampinKiller24**
Adds an alarm to the INDRA reactor that works similarly to the SM reactor alarm. Except less annoying since it doesn't start until it's 50% unstable.

### CODE REVIEW

**Comment by CampinKiller24** [comment]
!review

File: code/modules/power/fusion/core/core_field.dm [hunk 1]
```diff
@@ -45,6 +47,14 @@
 
 	var/animating_ripple = FALSE
 
+	var/obj/item/device/radio/radio
+	var/safe_alert = "INDRA core stabilizing."
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:51]:
```suggestion
	var/safe_alert = "NOTICE: INDRA reactor stabilizing."
```

File: code/modules/power/fusion/core/core_field.dm [hunk 2]
```diff
@@ -45,6 +47,14 @@
...
+	var/safe_warned = 0
+	var/public_alert = 0
+	var/warning_alert = "Attention! INDRA core instability rising!"
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:54]:
```suggestion
	var/warning_alert = "WARNING: INDRA reactor destabilizing!"
```

File: code/modules/power/fusion/core/core_field.dm [hunk 3]
```diff
@@ -45,6 +47,14 @@
...
+	var/emergency_alert = "ALERT! INDRA CORE MELTDOWN IMMINENT!"
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:55]:
```suggestion
	var/emergency_alert = "DANGER: INDRA REACTOR MELTDOWN IMMINENT!"
```

File: code/modules/power/fusion/core/core_field.dm [hunk 4]
```diff
@@ -271,6 +288,48 @@
 			Ripple(wave_size, ripple_radius)
 	return
 
+/obj/effect/fusion_em_field/proc/warning()
+	var/unstable = round(percent_unstable * 100)
+	var/alert_msg = " Instability at [unstable]%"
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:293]:
```suggestion
	var/alert_msg = " Instability at [unstable]%."
```

File: code/modules/power/fusion/core/core_field.dm [hunk 5]
```diff
@@ -271,6 +288,48 @@
...
+		if(percent_unstable >= percent_unstable_archive)
+			if(percent_unstable < 0.7)
+				alert_msg = warning_alert + alert_msg
+				lastwarning = world.timeofday
+				safe_warned = 0
+			else if(percent_unstable < 0.9)
+				alert_msg = emergency_alert + alert_msg
+				lastwarning = world.timeofday  - WARNING_DELAY * 4
+			else if(percent_unstable > 0.9)
+				lastwarning = world.timeofday  - WARNING_DELAY * 4
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:305]:
```suggestion
				lastwarning = world.timeofday - WARNING_DELAY * 4
			else if(percent_unstable > 0.9)
				lastwarning = world.timeofday - WARNING_DELAY * 4
```

File: code/modules/power/fusion/core/core_field.dm [hunk 6]
```diff
@@ -271,6 +288,48 @@
...
+			alert_msg = null
+	else
+		alert_msg = null
+	
+	if(alert_msg)
+		radio.autosay(alert_msg, "INDRA Reactor Monitor", "Engineering")
+
+		if((percent_unstable > 0.9) && !public_alert)
+			alert_msg = null
+			radio.autosay("ALERT! INDRA REACTOR CORE MELTDOWN IMMINENT!", "INDRA Reactor Monitor")
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:323]:
```suggestion
			radio.autosay(emergency_alert, "INDRA Reactor Monitor")
```

File: code/modules/power/fusion/core/core_field.dm [hunk 7]
```diff
@@ -271,6 +288,48 @@
...
+/obj/effect/fusion_em_field/proc/warning()
+	var/unstable = round(percent_unstable * 100)
+	var/alert_msg = " Instability at [unstable]%"
+
+	if(percent_unstable > 0.5)
+		if(percent_unstable >= percent_unstable_archive)
+			if(percent_unstable < 0.7)
+				alert_msg = warning_alert + alert_msg
+				lastwarning = world.timeofday
+				safe_warned = 0
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:300]:
```suggestion
				safe_warned = FALSE
```

File: code/modules/power/fusion/core/core_field.dm [hunk 8]
```diff
@@ -271,6 +288,48 @@
...
+			else if(percent_unstable < 0.9)
+				alert_msg = emergency_alert + alert_msg
+				lastwarning = world.timeofday  - WARNING_DELAY * 4
+			else if(percent_unstable > 0.9)
+				lastwarning = world.timeofday  - WARNING_DELAY * 4
+				alert_msg = emergency_alert + alert_msg
+			else
+				alert_msg = null
+		else if(!safe_warned)
+			safe_warned = 1
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:310]:
```suggestion
			safe_warned = TRUE
```

File: code/modules/power/fusion/core/core_field.dm [hunk 9]
```diff
@@ -271,6 +288,48 @@
...
+	else
+		alert_msg = null
+	
+	if(alert_msg)
+		radio.autosay(alert_msg, "INDRA Reactor Monitor", "Engineering")
+
+		if((percent_unstable > 0.9) && !public_alert)
+			alert_msg = null
+			radio.autosay("ALERT! INDRA REACTOR CORE MELTDOWN IMMINENT!", "INDRA Reactor Monitor")
+			public_alert = 1
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:324]:
```suggestion
			public_alert = TRUE
```

File: code/modules/power/fusion/core/core_field.dm [hunk 10]
```diff
@@ -271,6 +288,48 @@
...
+				for(var/mob/M in player_list)
+					var/turf/T = get_turf(M)
+					if(T && !istype(M, /mob/abstract/new_player) && !isdeaf(M))
+						sound_to(M, 'sound/effects/nuclearsiren.ogg')
+		else if(safe_warned && public_alert)
+			radio.autosay(alert_msg, "INDRA Reactor Monitor")
+			public_alert = 0
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:331]:
```suggestion
			public_alert = FALSE
```

File: code/modules/power/fusion/core/core_field.dm [hunk 11]
```diff
@@ -45,6 +47,14 @@
 
 	var/animating_ripple = FALSE
 
+	var/obj/item/device/radio/radio
+	var/safe_alert = "NOTICE: INDRA reactor stabilizing."
+	var/safe_warned = 0
+	var/public_alert = 0
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:53]:
```suggestion
	var/safe_warned = FALSE
	var/public_alert = FALSE
```

File: code/modules/power/fusion/core/core_field.dm [hunk 12]
```diff
@@ -271,6 +288,48 @@
...
+		radio.autosay(alert_msg, "INDRA Reactor Monitor", "Engineering")
+
+		if((percent_unstable > 0.9) && !public_alert)
+			alert_msg = null
+			radio.autosay(emergency_alert, "INDRA Reactor Monitor")
+			public_alert = TRUE
+				for(var/mob/M in player_list)
+					var/turf/T = get_turf(M)
+					if(T && !istype(M, /mob/abstract/new_player) && !isdeaf(M))
+						sound_to(M, 'sound/effects/nuclearsiren.ogg')
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:328]:
```suggestion
			for(var/mob/M in player_list)
				var/turf/T = get_turf(M)
				if(T && !istype(M, /mob/abstract/new_player) && !isdeaf(M))
					sound_to(M, 'sound/effects/nuclearsiren.ogg')
```

File: code/modules/power/fusion/core/core_field.dm [hunk 13]
```diff
@@ -2,6 +2,7 @@
 #define FUSION_INSTABILITY_DIVISOR 50000
 #define FUSION_RUPTURE_THRESHOLD   10000
 #define FUSION_REACTANT_CAP        10000
+#define WARNING_DELAY 20
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:5]:
```suggestion
#define FUSION_WARNING_DELAY 20
```
this is already used in the supermatter. you can't use the same define names.

File: code/modules/power/fusion/core/core_field.dm [hunk 14]
```diff
@@ -202,13 +213,19 @@
 			radiation += radiate
 
 	check_instability()
+
+	if(percent_unstable > 0.5 && (percent_unstable >= percent_unstable_archive))
+		if((world.timeofday - lastwarning) >= WARNING_DELAY * 10)
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:218]:
```suggestion
		if((world.timeofday - lastwarning) >= FUSION_WARNING_DELAY * 10)
```

File: code/modules/power/fusion/core/core_field.dm [hunk 15]
```diff
@@ -271,6 +288,48 @@
...
+		if(percent_unstable >= percent_unstable_archive)
+			if(percent_unstable < 0.7)
+				alert_msg = warning_alert + alert_msg
+				lastwarning = world.timeofday
+				safe_warned = FALSE
+			else if(percent_unstable < 0.9)
+				alert_msg = emergency_alert + alert_msg
+				lastwarning = world.timeofday - WARNING_DELAY * 4
+			else if(percent_unstable > 0.9)
+				lastwarning = world.timeofday - WARNING_DELAY * 4
```
Comment (SleepyGemmy) [inline code/modules/power/fusion/core/core_field.dm:305]:
```suggestion
				lastwarning = world.timeofday - FUSION_WARNING_DELAY * 4
			else if(percent_unstable > 0.9)
				lastwarning = world.timeofday - FUSION_WARNING_DELAY * 4
```

**Review by Alberyk — APPROVED** [review]

**Review by SleepyGemmy — APPROVED** [review]

**Pull request merged**